src/app/routes/getInitialData/utils/fetchData.js

```diff
@@ -0,0 +1,34 @@
+import 'isomorphic-fetch';
+import nodeLogger from '../../../helpers/logger.node';
+import preprocess from '../../../dataValidator/preprocessor';
+
+const logger = nodeLogger(__filename);
+const upstreamStatusCodesToPropagate = [200, 404];
+
+const fetchData = async ({ url, preprocessorRules }) => {
+  let data;
+  let status;
+
+  try {
+    const response = await fetch(url);
+
+    status = response.status; // eslint-disable-line prefer-destructuring
+
+    if (status === 200) {
+      data = await response.json();
+      data = preprocess(data, preprocessorRules);
```

Comment:
Nitpick: I wondered if the `preprocess` step should be called somewhere else to keep this util module just focused on fetching/status concerns and not create an interdependency on the validation phase. What do you think?

Reply:
@jamesdonoh I think it was put in here as if it fails, its a symptom of an invalid response from the fetch, and gets caught in the catch of the try/catch. We could maybe raise an issue to move it if you have a suggestion of wheres a better option :)

Reply:
Ah. It's not very obvious from the code that `preprocess` can throw errors that we catch and want to handle as 502s, is it?

Reply:
Perhaps not, is this ok to merge? We can raise an issue to move it, but doing so now would change the behaviour

Reply:
I'm not sure I fully follow the conversation here. @jamesdonoh are you suggesting something like:
```
const getData() => {
  try {
    fetchData();
    // the data fetched successfully but is a bad status
    if (status !== error) {
      return data;
    }
  } catch {
    throw Error; // the data fetching failed
  }
}

// this could live in a different function or file
export const getProcessedData = preprocess(getData());
```

Reply:
I'm not really suggesting a specific structure, I'm saying that `fetchData` is currently both fetching and preprocessing data, and that the fact that it also does error handling on the preprocessor is not obvious to a first-time reader. But overall the PR is clearly a better factoring of what's there already and I don't mind if those concerns get addressed separately

If we were using JSDoc I'd suggest we tagged the implementation of preprocess with `@throws`, but we aren't. A brief inline comment might help.